### PR TITLE
AEROGEAR-2209 Add Encryption Check

### PR DESCRIPTION
**Pull request by ciaranRoche**
## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-2209

## Description

We created a function that allows the end developer to check if the devices file system is encrypted

## Progress

- [x] Add Check
- [x] Add Unit Tests
- [x] Add Javadoc
- [x] Add Check to Example App

### CODE REVIEW

File: auth/src/main/java/org/aerogear/mobile/security/checks/EncryptionCheck.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+public class EncryptionCheck extends AbstractSecurityCheck{
+    /**
+     * Check if a devices filesystem is encrypted
+     *
+     * @param context Context to be used by the check.
+     * @return <code>true</code> if the encryption have been enabled on the device.
+     */
+    @Override
+    protected boolean execute(@NonNull Context context) {
```
Comment (ziccardi) [inline auth/src/main/java/org/aerogear/mobile/security/checks/EncryptionCheck.java]: make parameter final

File: example/src/main/java/org/aerogear/mobile/example/ui/SecurityServiceFragment.java [hunk 2]
```diff
@@ -178,7 +178,10 @@ public void detectBackupEnabled(Map<String, SecurityCheckResult> results) {
      */
     public void detectDeviceEncryptionStatus(Map<String, SecurityCheckResult> results) {
         totalTests++;
-        //TODO: add check
+        SecurityCheckResult result = securityService.check(SecurityCheckType.HAS_ENCRYPTION_ENABLED);
+        if (result.passed()){
+            setDetected(deviceEncrypted, R.string.device_encrypted_positive);
```
Comment (aidenkeating) [inline example/src/main/java/org/aerogear/mobile/example/ui/SecurityServiceFragment.java]: Is this correct, isn't encryption enabled a good thing? At the moment I'm seeing that encryption is enabled but it's in red writing.

```java
if (!result.passed()) {
    setDetected(deviceEncrypted, R.string.device_encrypted_negative);
}
```

would make more sense to me.

**Comment by ciaranRoche** [comment]
@ziccardi @aidenkeating @TommyJ1994 hey guys made those changes, if ye could have another look when you get a chance.

File: example/src/main/java/org/aerogear/mobile/example/ui/SecurityServiceFragment.java [hunk 3]
```diff
@@ -154,14 +147,6 @@ public void detectEmulator(Map<String, SecurityCheckResult> results) {
         }
     }
 
-    /**
-     * Detect if a hooking framework application is installed on the device
-     */
-    public void detectHookingFramework(Map<String, SecurityCheckResult> results) {
```
Comment (aidenkeating) [inline example/src/main/java/org/aerogear/mobile/example/ui/SecurityServiceFragment.java]: Do we intend on adding this check at all or is it not being added?

Comment (TommyJ1994) [reply example/src/main/java/org/aerogear/mobile/example/ui/SecurityServiceFragment.java]: @ciaranRoche @aidenkeating I think we can drop this one for now. It was really a POC for the mobile security project.

**Pull request merged**